src/client/Microsoft.Identity.Client/TelemetryCore/TelemetryManager.cs

```diff
@@ -137,6 +147,93 @@ public void Flush(string correlationId)
             Callback?.Invoke(eventsToFlush.Cast<Dictionary<string, string>>().ToList());
         }
 
+        public string FetchAndResetPreviousHttpTelemetryContent()
+        {
+            if (!string.IsNullOrEmpty(_previousTelemetryPayload.ApiId))
+            {
+                string csv = CreateCSVForPreviousHttpTelem();
+                _currentTelemetryPayload.ResetLastErrorCode();
+                return csv;
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
+        public string FetchAndResetCurrentHttpTelemetryContent()
+        {
+            if (!string.IsNullOrEmpty(_currentTelemetryPayload.ApiId))
+            {               
+                string csv = CreateCSVForCurrentHttpTelem();
+                _currentTelemetryPayload.ResetLastErrorCode();
+                return csv;
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
+        private string CreateCSVForPreviousHttpTelem()
+        {
+            // csv expected format:
+            // 1|api_id, correlation_id, last_error_code
+            string[] myValues = new string[] {
+                _previousTelemetryPayload.ApiId,
+                _previousTelemetryPayload.CorrelationId,
+                _previousTelemetryPayload.LastErrorCode};
+
+            string csvString = string.Join(",", myValues);
+            csvString = TelemetryConstants.HttpTelemetrySchemaVersion1 + csvString;
+            return csvString;
+        }
+
+        private string CreateCSVForCurrentHttpTelem()
+        {
+            // csv expected format:
+            // 1|api_id|platform_config
+            string csvString = TelemetryConstants.HttpTelemetrySchemaVersion1 + _currentTelemetryPayload.ApiId;
+            return csvString;
+        }
+
+        private void ProcessEventsForCurrentHttpTelemetryContent(EventBase events)
+        {
+            foreach (var ev in events)
+            {
+                if (string.Equals(ev.Key, MsalTelemetryBlobEventNames.ApiIdConstStrKey))
+                {
+                    _currentTelemetryPayload.ApiId = ev.Value;
+                }
+
+                if (string.Equals(ev.Key, MsalTelemetryBlobEventNames.MsalCorrelationIdConstStrKey))
+                {
+                    _currentTelemetryPayload.CorrelationId = ev.Value;
+                }
+            }
+        }
+
+        private void ProcessEventsForPreviousHttpTelemetryContent(EventBase events)
+        {
+            foreach (var ev in events)
+            {
+                if (string.Equals(ev.Key, MsalTelemetryBlobEventNames.ApiIdConstStrKey))
+                {
+                    _previousTelemetryPayload.ApiId = ev.Value;
+                }
+
+                if (string.Equals(ev.Key, MsalTelemetryBlobEventNames.ApiErrorCodeConstStrKey))
```

Comment:
I don't understand where this data is written since you've just defined this key, but you are not using it anywhere.

Reply:
not sure i follow.

---
In reply to: [325790729](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1389#discussion_r325790729) [](ancestors = 325790729)